pkg/apis/storage/validation/validation.go

```diff
@@ -253,8 +253,20 @@ func validateAllowedTopologies(topologies []api.TopologySelectorTerm, fldPath *f
 		allErrs = append(allErrs, field.Forbidden(fldPath, "field is disabled by feature-gate DynamicProvisioningScheduling"))
 	}
 
+	rawTopologies := make([]map[string]sets.String, len(topologies))
 	for i, term := range topologies {
-		allErrs = append(allErrs, apivalidation.ValidateTopologySelectorTerm(term, fldPath.Index(i))...)
+		idxPath := fldPath.Index(i)
+		exprMap, termErrs := apivalidation.ValidateTopologySelectorTerm(term, fldPath.Index(i))
+		allErrs = append(allErrs, termErrs...)
+
+		// TODO (verult) consider improving runtime
+		for _, t := range rawTopologies {
+			if reflect.DeepEqual(exprMap, t) {
```

Comment:
reflect.DeepEqual or semantic.DeepEqual? (reflect differentiates between nil and [], semantic does not)

Reply:
Neat, in this case reflect.DeepEqual is OK but semantic.DeepEqual is safer. Will update